integration-libs/cdc/root/service/cdc-js.service.ts

```diff
@@ -365,6 +435,151 @@ export class CdcJsService implements OnDestroy {
+                profile: {
+                  email: newEmail,
+                },
+              }).pipe(
+                tap({
+                  next: () =>
+                    this.userProfileFacade.update({ uid: newEmail }).pipe(
+                      tap({
+                        error: (error) => of(error),
+                        complete: () => {
+                          this.auth.coreLogout();
+                          this.invokeAPI('accounts.logout', {});
+                        },
+                      })
+                    ),
+                })
+              )
+            ),
+            catchError((error) => of(error))
+          );
+        })
+      );
+    }
+  }
+
+  /**
+   * Obtain the email of the currently logged in user
+   * @returns emailID of the loggedIn user
+   */
+  protected getLoggedInUserEmail(): Observable<User> {
+    return this.userProfileFacade.get().pipe(
+      filter((user): user is User => Boolean(user))
+      take(1),
+    );
+  }
+
+  /**
+   * Trigger CDC address update.
+   *
+   * @param address
+   */
+  updateAddressWithoutScreenSet(
+    formattedAddress: string,
+    zipCode?: string,
+    city?: string,
+    country?: string
+  ): Observable<{ status: string }> {
+    if (!formattedAddress || formattedAddress?.length === 0) {
+      return throwError({ errorMessage: 'No address provided' });
+    } else {
+      let profileObj = {
+        address: formattedAddress,
+        ...(city && { city: city }),
+        ...(country && { country: country }),
+        ...(zipCode && { zip: zipCode }),
+      };
+      return this.invokeAPI('accounts.setAccountInfo', {
+        profile: profileObj,
+      });
+    }
+  }
+
+  /**
+   * Obtain the CDC SDK Method from the input method name as string
+   * @param methodName
+   * @returns CDC SDK Function
+   */
+  protected getSdkFunctionFromName(
+    methodName: string
+  ): (payload: Object) => void {
+    //accounts.setAccountInfo or accounts.b2b.openDelegatedAdmin
+    let nestedMethods = methodName.split('.');
+    let cdcAPI: any = this.gigyaSDK;
+    nestedMethods.forEach((method) => {
+      if (cdcAPI && cdcAPI.hasOwnProperty(method)) {
+        cdcAPI = cdcAPI[method];
+      }
+    });
+
+    return cdcAPI;
+  }
+
+  /**
+   * Invoke the CDC SDK Method and convert the callback to an Observable
+   * @param methodName - method to be invoked
+   * @param payload - Object payload
+   * @returns - Observable with the response
+   */
+  protected invokeAPI(
+    methodName: string,
+    payload: Object
+  ): Observable<{ status: string }> {
+    return new Observable<{ status: string }>((result) => {
+      let actualAPI = this.getSdkFunctionFromName(methodName);
+
+      actualAPI({
+        ...payload,
+        callback: (response: any) => {
+          if (response?.status === 'OK') {
+            result.next(response);
+            result.complete();
+          } else {
+            result.error(response);
+          }
```

Comment:
As Mateusz suggested:
To fix broken change detection after user login, let's wrap it withing ngZone

```suggestion
          this.zone.run(()=> {
            if (response?.status === 'OK') {
              result.next(response);
              result.complete();
            } else {
              result.error(response);
            }
          });
```

Reply:
noted.

Reply:
It looks like we have to wrap within ngZone all the callbacks that we pass to CDC SDK, that are later invoked by CDC SDK. It feels to me like the correct solution to our problem.

But to be very honest with you, I don’t know _why exactly_ we have to do it. To my knowledge about NgZone, we should not need to do that. So either I’m lacking some knowledge about Angular’s NgZone OR maybe there is something very specific to the CDC SDK (it’s internal implementation) that NgZone “doesn’t work” inside of CDC SDK out of the box. 

A bit of theory:
ZoneJS (which is used by NgZone) is monkey-patching all WebAPI for async operations (like http calls, setTimeout, etc.) and thanks to that ZoneJS can track all async operations happening in the app. When some async operation is completed, ZoneJS intercepts this fact. Then NgZone triggers a Change Detection in the Angular app.

Our problem looks like - to me at the first sight - like the ZoneJS is not able to intercept the completion of async operation that was triggered from within of CDC SDK (like completion of a http call to CDC API). And to my understanding that’s why the Change Detection was not triggered in out Angular app, causing the UI not to update. I deduce it from the fact that invoking the `callback` code explicitly within the NgZone fixes the problem - the Change Detection is triggered correctly then.

It’s not crucial for 6.0 to investigate _why_ CDC SDK is not running within NgZone by default as I would expect. But it’s very curious and after 6.0 I’d be happy to check it. Perhaps investigation of the source code of CDC SDK could give me some insight...

cc @armanozak @Matejk00 @SherwinVarghese